query/compilation.rs

```diff
@@ -31,6 +40,11 @@ pub enum CompiledStage {
     Match(MatchProgram),
     Insert(InsertProgram),
     Delete(DeleteProgram),
+
+    Filter(SelectProgram),
```

Comment:
Almost got all the renames! Some more here --
I'll merge the PR and you can roll the change into `reduce` operations.